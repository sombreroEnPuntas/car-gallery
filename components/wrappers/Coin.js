import styled from 'styled-components'
import { colors } from '.'

const Coin = styled.div`
  width: 21px;
  height: 12px;
  background: ${colors.coin.main};
  box-shadow:
    /* 
    * Inner Background Color
    */ 0 -6px 0 -2px
      ${colors.coin.main},
    0 6px 0 -2px ${colors.coin.main}, 0 -12px 0 -4px ${colors.coin.main},
    0 12px 0 -4px ${colors.coin.main}, 0 -18px 0 -8px ${colors.coin.main},
    0 18px 0 -8px ${colors.coin.main}, 0 -24px 0 -12px ${colors.coin.main},
    0 24px 0 -12px ${colors.coin.main},
    /* 
    * Inner Border
    */ -2px 0 0 0 ${colors.coin.border},
    2px 0 0 0 ${colors.coin.border}, 0 -4px 0 0 ${colors.coin.border},
    0 4px 0 0 ${colors.coin.border}, 0 -10px 0 -2px ${colors.coin.border},
    0 10px 0 -2px ${colors.coin.border}, 0 -14px 0 -4px ${colors.coin.border},
    0 14px 0 -4px ${colors.coin.border}, 0 -20px 0 -8px ${colors.coin.border},
    0 20px 0 -8px ${colors.coin.border}, 0 -26px 0 -12px ${colors.coin.border},
    0 26px 0 -12px ${colors.coin.border},
    /* 
    * Middle Border 
    */ -4px 0 0 0 ${colors.coin.accent},
    4px 0 0 0 ${colors.coin.accent}, 0 -2px 0 2px ${colors.coin.accent},
    0 2px 0 2px ${colors.coin.accent}, 0 -8px 0 0px ${colors.coin.accent},
    0 8px 0 0px ${colors.coin.accent}, 0 -12px 0 -2px ${colors.coin.accent},
    0 12px 0 -2px ${colors.coin.accent}, 0 -16px 0 -4px ${colors.coin.accent},
    0 16px 0 -4px ${colors.coin.accent}, 0 -22px 0 -8px ${colors.coin.accent},
    0 22px 0 -8px ${colors.coin.accent}, 0 -28px 0 -12px ${colors.coin.accent},
    0 28px 0 -12px ${colors.coin.accent},
    /*
    * Outer Border
    */ -6px 0 0 0 ${colors.coin.border},
    6px 0 0 0 ${colors.coin.border}, 0 0 0 4px ${colors.coin.border},
    0 -6px 0 2px ${colors.coin.border}, 0 6px 0 2px ${colors.coin.border},
    0 -10px 0 0 ${colors.coin.border}, 0 10px 0 0 ${colors.coin.border},
    0 -14px 0 -2px ${colors.coin.border}, 0 14px 0 -2px ${colors.coin.border},
    0 -18px 0 -4px ${colors.coin.border}, 0 18px 0 -4px ${colors.coin.border},
    0 -24px 0 -8px ${colors.coin.border}, 0 24px 0 -8px ${colors.coin.border},
    0 -30px 0 -12px ${colors.coin.border}, 0 30px 0 -12px ${colors.coin.border};
`

export default Coin
