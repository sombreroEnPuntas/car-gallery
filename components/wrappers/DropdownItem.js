import styled from 'styled-components'
import { fonts } from '.'

export const DropdownItem = styled.li`
  align-items: center;
  display: flex;
  height: ${fonts.lineHeight};
  justify-content: center;
  margin: 0;
  overflow: hidden;
  padding: 0;
  text-align: center;
  text-decoration: none;
  text-overflow: ellipsis;
  text-transform: capitalize;
  white-space: nowrap;
  width: 100%;
`

export default DropdownItem
