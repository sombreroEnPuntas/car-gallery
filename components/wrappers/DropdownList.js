import styled from 'styled-components'
import { colors, fonts } from '.'

const DropdownList = styled.ul`
  background-color: white;
  border: solid 2px;
  border-top: 0;
  box-shadow: 4px 0 ${colors['black'].shadow}, 0 -4px ${colors['black'].shadow},
    -4px 0 ${colors['black'].shadow}, 0 4px ${colors['black'].shadow};
  list-style: none;
  margin: 0 4px;
  max-height: calc(6 * ${fonts.lineHeight});
  overflow-y: hidden;
  padding: 0;
  position: absolute;
  top: 0;
  width: calc(100% - 12px);
`

export default DropdownList
