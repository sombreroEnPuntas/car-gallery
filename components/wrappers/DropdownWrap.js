import styled from 'styled-components'

const DropdownWrap = styled.div`
  position: relative;
  width: 100%;
  z-index: 1;
`

export default DropdownWrap
