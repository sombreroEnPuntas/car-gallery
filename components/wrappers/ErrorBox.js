import styled from 'styled-components'
import { colors } from '.'

const ErrorBox = styled.pre`
  background-color: ${colors.error.background};
  box-shadow: 4px 0 ${colors.error.shadow}, 0 -4px ${colors.error.shadow},
    -4px 0 ${colors.error.shadow}, 0 4px ${colors.error.shadow};
  display: block;
  margin: 0.5em 0;
  overflow-x: auto;
  padding: 0.5em;
  white-space: pre-wrap;
`

export default ErrorBox
