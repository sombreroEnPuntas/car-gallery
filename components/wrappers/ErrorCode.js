import styled from 'styled-components'
import { colors, fonts } from '.'

const ErrorCode = styled.code`
  color: ${colors.error.outline};
  font-family: ${fonts.fontFamily};
  font-size: 0.65em;
  font-weight: bold;
`

export default ErrorCode
