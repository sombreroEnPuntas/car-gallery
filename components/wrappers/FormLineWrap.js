import styled from 'styled-components'

const FormLineWrap = styled.div`
  display: flex;
  width: 100%;
`

export default FormLineWrap
