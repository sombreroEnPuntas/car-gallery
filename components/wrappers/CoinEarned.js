import styled from 'styled-components'
import { animations } from '.'
import Coin from './Coin'

const CoinEarned = styled(Coin)`
  animation: ${animations.rotateIn} 1.4s ease-out forwards,
    ${animations.fadeOut} 2s ease-out forwards;

  position: absolute;
`

export default CoinEarned
