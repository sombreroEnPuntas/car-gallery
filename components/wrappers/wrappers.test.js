// @flow
import React from 'react'
// Generate snapshots including the styles
import 'jest-styled-components'

import { shallow, mount } from 'enzyme'

import Coin from './Coin'
import CoinEarned from './CoinEarned'
import DropdownItem from './DropdownItem'
import DropdownList from './DropdownList'
import DropdownWrap from './DropdownWrap'
import ErrorBox from './ErrorBox'
import ErrorCode from './ErrorCode'
import FormLineWrap from './FormLineWrap'
import Input from './Input'
import MessageBox from './MessageBox'
import Modal from './Modal'
import PageWrapper from './PageWrapper'
import ThreeBlinking from './ThreeBlinking'

const wrappers = {
  Coin,
  CoinEarned,
  DropdownItem,
  DropdownList,
  DropdownWrap,
  ErrorBox,
  ErrorCode,
  FormLineWrap,
  Input,
  MessageBox,
  Modal,
  PageWrapper,
  ThreeBlinking,
}

const getProps = customProps => ({
  ...customProps,
})

describe.each`
  displayName        | props
  ${'Coin'}          | ${null}
  ${'CoinEarned'}    | ${null}
  ${'DropdownItem'}  | ${null}
  ${'DropdownList'}  | ${null}
  ${'DropdownWrap'}  | ${null}
  ${'ErrorBox'}      | ${null}
  ${'ErrorCode'}     | ${null}
  ${'FormLineWrap'}  | ${null}
  ${'Input'}         | ${{ status: 'error' }}
  ${'Input'}         | ${{ status: 'success' }}
  ${'Input'}         | ${{ status: null }}
  ${'MessageBox'}    | ${null}
  ${'Modal'}         | ${{ open: false }}
  ${'Modal'}         | ${{ open: true }}
  ${'PageWrapper'}   | ${null}
  ${'ThreeBlinking'} | ${null}
`(`TestedComponent is $displayName and`, ({ displayName, props }) => {
  it(`matches snapshot`, () => {
    const TestedComponent = wrappers[displayName]
    const wrapper =
      props === null
        ? shallow(<TestedComponent />)
        : mount(<TestedComponent {...getProps(props)} />)

    expect(wrapper).toMatchSnapshot()
  })
})
