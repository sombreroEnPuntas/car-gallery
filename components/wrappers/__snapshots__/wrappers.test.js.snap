// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TestedComponent is Coin and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "Coin-lmzeq1-0",
        "isStatic": true,
        "rules": Array [
          "width:21px;height:12px;background:",
          "gold",
          ";box-shadow:0 -6px 0 -2px ",
          "gold",
          ",0 6px 0 -2px ",
          "gold",
          ",0 -12px 0 -4px ",
          "gold",
          ",0 12px 0 -4px ",
          "gold",
          ",0 -18px 0 -8px ",
          "gold",
          ",0 18px 0 -8px ",
          "gold",
          ",0 -24px 0 -12px ",
          "gold",
          ",0 24px 0 -12px ",
          "gold",
          ",-2px 0 0 0 ",
          "black",
          ",2px 0 0 0 ",
          "black",
          ",0 -4px 0 0 ",
          "black",
          ",0 4px 0 0 ",
          "black",
          ",0 -10px 0 -2px ",
          "black",
          ",0 10px 0 -2px ",
          "black",
          ",0 -14px 0 -4px ",
          "black",
          ",0 14px 0 -4px ",
          "black",
          ",0 -20px 0 -8px ",
          "black",
          ",0 20px 0 -8px ",
          "black",
          ",0 -26px 0 -12px ",
          "black",
          ",0 26px 0 -12px ",
          "black",
          ",-4px 0 0 0 ",
          "goldenrod",
          ",4px 0 0 0 ",
          "goldenrod",
          ",0 -2px 0 2px ",
          "goldenrod",
          ",0 2px 0 2px ",
          "goldenrod",
          ",0 -8px 0 0px ",
          "goldenrod",
          ",0 8px 0 0px ",
          "goldenrod",
          ",0 -12px 0 -2px ",
          "goldenrod",
          ",0 12px 0 -2px ",
          "goldenrod",
          ",0 -16px 0 -4px ",
          "goldenrod",
          ",0 16px 0 -4px ",
          "goldenrod",
          ",0 -22px 0 -8px ",
          "goldenrod",
          ",0 22px 0 -8px ",
          "goldenrod",
          ",0 -28px 0 -12px ",
          "goldenrod",
          ",0 28px 0 -12px ",
          "goldenrod",
          ",-6px 0 0 0 ",
          "black",
          ",6px 0 0 0 ",
          "black",
          ",0 0 0 4px ",
          "black",
          ",0 -6px 0 2px ",
          "black",
          ",0 6px 0 2px ",
          "black",
          ",0 -10px 0 0 ",
          "black",
          ",0 10px 0 0 ",
          "black",
          ",0 -14px 0 -2px ",
          "black",
          ",0 14px 0 -2px ",
          "black",
          ",0 -18px 0 -4px ",
          "black",
          ",0 18px 0 -4px ",
          "black",
          ",0 -24px 0 -8px ",
          "black",
          ",0 24px 0 -8px ",
          "black",
          ",0 -30px 0 -12px ",
          "black",
          ",0 30px 0 -12px ",
          "black",
          ";",
        ],
      },
      "displayName": "Coin",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "Coin-lmzeq1-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is CoinEarned and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "CoinEarned-av3i5n-0",
        "isStatic": true,
        "rules": Array [
          "width:21px;height:12px;background:",
          "gold",
          ";box-shadow:0 -6px 0 -2px ",
          "gold",
          ",0 6px 0 -2px ",
          "gold",
          ",0 -12px 0 -4px ",
          "gold",
          ",0 12px 0 -4px ",
          "gold",
          ",0 -18px 0 -8px ",
          "gold",
          ",0 18px 0 -8px ",
          "gold",
          ",0 -24px 0 -12px ",
          "gold",
          ",0 24px 0 -12px ",
          "gold",
          ",-2px 0 0 0 ",
          "black",
          ",2px 0 0 0 ",
          "black",
          ",0 -4px 0 0 ",
          "black",
          ",0 4px 0 0 ",
          "black",
          ",0 -10px 0 -2px ",
          "black",
          ",0 10px 0 -2px ",
          "black",
          ",0 -14px 0 -4px ",
          "black",
          ",0 14px 0 -4px ",
          "black",
          ",0 -20px 0 -8px ",
          "black",
          ",0 20px 0 -8px ",
          "black",
          ",0 -26px 0 -12px ",
          "black",
          ",0 26px 0 -12px ",
          "black",
          ",-4px 0 0 0 ",
          "goldenrod",
          ",4px 0 0 0 ",
          "goldenrod",
          ",0 -2px 0 2px ",
          "goldenrod",
          ",0 2px 0 2px ",
          "goldenrod",
          ",0 -8px 0 0px ",
          "goldenrod",
          ",0 8px 0 0px ",
          "goldenrod",
          ",0 -12px 0 -2px ",
          "goldenrod",
          ",0 12px 0 -2px ",
          "goldenrod",
          ",0 -16px 0 -4px ",
          "goldenrod",
          ",0 16px 0 -4px ",
          "goldenrod",
          ",0 -22px 0 -8px ",
          "goldenrod",
          ",0 22px 0 -8px ",
          "goldenrod",
          ",0 -28px 0 -12px ",
          "goldenrod",
          ",0 28px 0 -12px ",
          "goldenrod",
          ",-6px 0 0 0 ",
          "black",
          ",6px 0 0 0 ",
          "black",
          ",0 0 0 4px ",
          "black",
          ",0 -6px 0 2px ",
          "black",
          ",0 6px 0 2px ",
          "black",
          ",0 -10px 0 0 ",
          "black",
          ",0 10px 0 0 ",
          "black",
          ",0 -14px 0 -2px ",
          "black",
          ",0 14px 0 -2px ",
          "black",
          ",0 -18px 0 -4px ",
          "black",
          ",0 18px 0 -4px ",
          "black",
          ",0 -24px 0 -8px ",
          "black",
          ",0 24px 0 -8px ",
          "black",
          ",0 -30px 0 -12px ",
          "black",
          ",0 30px 0 -12px ",
          "black",
          ";",
          "animation:",
          Keyframes {
            "id": "sc-keyframes-isQrCp",
            "inject": [Function],
            "name": "isQrCp",
            "rules": Array [
              "@-webkit-keyframes isQrCp{0%{opacity:0;-webkit-transform:rotateY(0) translateY(100%);-ms-transform:rotateY(0) translateY(100%);transform:rotateY(0) translateY(100%);}100%{opacity:1;-webkit-transform:rotateY(1800deg) translateY(0);-ms-transform:rotateY(1800deg) translateY(0);transform:rotateY(1800deg) translateY(0);}}",
              "@keyframes isQrCp{0%{opacity:0;-webkit-transform:rotateY(0) translateY(100%);-ms-transform:rotateY(0) translateY(100%);transform:rotateY(0) translateY(100%);}100%{opacity:1;-webkit-transform:rotateY(1800deg) translateY(0);-ms-transform:rotateY(1800deg) translateY(0);transform:rotateY(1800deg) translateY(0);}}",
            ],
            "toString": [Function],
          },
          " 1.4s ease-out forwards,",
          Keyframes {
            "id": "sc-keyframes-biVnR",
            "inject": [Function],
            "name": "biVnR",
            "rules": Array [
              "@-webkit-keyframes biVnR{70%{opacity:1;}100%{opacity:0;}}",
              "@keyframes biVnR{70%{opacity:1;}100%{opacity:0;}}",
            ],
            "toString": [Function],
          },
          " 2s ease-out forwards;",
        ],
      },
      "displayName": "CoinEarned",
      "foldedComponentIds": Array [
        "Coin-lmzeq1-0",
      ],
      "render": [Function],
      "styledComponentId": "CoinEarned-av3i5n-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is DropdownItem and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "DropdownItem-pa8zp2-0",
        "isStatic": true,
        "rules": Array [
          "align-items:center;display:flex;height:",
          "1.5em",
          ";justify-content:center;margin:0;overflow:hidden;padding:0;text-align:center;text-decoration:none;text-overflow:ellipsis;text-transform:capitalize;white-space:nowrap;width:100%;",
        ],
      },
      "displayName": "DropdownItem",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "DropdownItem-pa8zp2-0",
      "target": "li",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is DropdownList and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "DropdownList-cryzk1-0",
        "isStatic": true,
        "rules": Array [
          "background-color:white;border:solid 2px;border-top:0;box-shadow:4px 0 ",
          "#212529",
          ",0 -4px ",
          "#212529",
          ",-4px 0 ",
          "#212529",
          ",0 4px ",
          "#212529",
          ";list-style:none;margin:0 4px;max-height:calc(6 * ",
          "1.5em",
          ");overflow-y:hidden;padding:0;position:absolute;top:0;width:calc(100% - 12px);",
        ],
      },
      "displayName": "DropdownList",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "DropdownList-cryzk1-0",
      "target": "ul",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is DropdownWrap and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "DropdownWrap-sc-7fme7b-0",
        "isStatic": true,
        "rules": Array [
          "position:relative;width:100%;z-index:1;",
        ],
      },
      "displayName": "DropdownWrap",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "DropdownWrap-sc-7fme7b-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is ErrorBox and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "ErrorBox-xbxs9c-0",
        "isStatic": true,
        "rules": Array [
          "background-color:",
          "rgba(231, 110, 85, 0.5)",
          ";box-shadow:4px 0 ",
          "#e76e55",
          ",0 -4px ",
          "#e76e55",
          ",-4px 0 ",
          "#e76e55",
          ",0 4px ",
          "#e76e55",
          ";display:block;margin:0.5em 0;overflow-x:auto;padding:0.5em;white-space:pre-wrap;",
        ],
      },
      "displayName": "ErrorBox",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "ErrorBox-xbxs9c-0",
      "target": "pre",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is ErrorCode and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "ErrorCode-sc-1u9g8cu-0",
        "isStatic": true,
        "rules": Array [
          "color:",
          "#ce372b",
          ";font-family:",
          "'Press Start 2P', monospace",
          ";font-size:0.65em;font-weight:bold;",
        ],
      },
      "displayName": "ErrorCode",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "ErrorCode-sc-1u9g8cu-0",
      "target": "code",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is FormLineWrap and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "FormLineWrap-sc-118xyvq-0",
        "isStatic": true,
        "rules": Array [
          "display:flex;width:100%;",
        ],
      },
      "displayName": "FormLineWrap",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "FormLineWrap-sc-118xyvq-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is Input and matches snapshot 1`] = `
.c0 {
  border: none;
  box-shadow: 0 4px #e76e55,0 -4px #e76e55,4px 0 #e76e55,-4px 0 #e76e55;
  font-family: 'Press Start 2P',monospace;
  line-height: calc(2 * 1.5em);
  margin: 4px;
  outline-color: #ce372b;
  padding: 0.5rem 1rem;
  text-transform: capitalize;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1.5em;
    line-height: 1.5em;
  }
}

<Input
  status="error"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Input-sc-1wcekai-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "border:none;box-shadow:0 4px ",
            [Function],
            ",0 -4px ",
            [Function],
            ",4px 0 ",
            [Function],
            ",-4px 0 ",
            [Function],
            ";font-family:",
            "'Press Start 2P', monospace",
            ";line-height:calc(2 * ",
            "1.5em",
            ");margin:4px;outline-color:",
            [Function],
            ";padding:0.5rem 1rem;text-transform:capitalize;width:100%;@media (min-width:",
            "768px",
            "){font-size:",
            "1.5em",
            ";line-height:",
            "1.5em",
            ";}",
          ],
        },
        "displayName": "Input",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Input-sc-1wcekai-0",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    status="error"
  >
    <input
      className="c0"
    />
  </StyledComponent>
</Input>
`;

exports[`TestedComponent is Input and matches snapshot 2`] = `
.c0 {
  border: none;
  box-shadow: 0 4px #92cc41,0 -4px #92cc41,4px 0 #92cc41,-4px 0 #92cc41;
  font-family: 'Press Start 2P',monospace;
  line-height: calc(2 * 1.5em);
  margin: 4px;
  outline-color: #76c442;
  padding: 0.5rem 1rem;
  text-transform: capitalize;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1.5em;
    line-height: 1.5em;
  }
}

<Input
  status="success"
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Input-sc-1wcekai-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "border:none;box-shadow:0 4px ",
            [Function],
            ",0 -4px ",
            [Function],
            ",4px 0 ",
            [Function],
            ",-4px 0 ",
            [Function],
            ";font-family:",
            "'Press Start 2P', monospace",
            ";line-height:calc(2 * ",
            "1.5em",
            ");margin:4px;outline-color:",
            [Function],
            ";padding:0.5rem 1rem;text-transform:capitalize;width:100%;@media (min-width:",
            "768px",
            "){font-size:",
            "1.5em",
            ";line-height:",
            "1.5em",
            ";}",
          ],
        },
        "displayName": "Input",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Input-sc-1wcekai-0",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    status="success"
  >
    <input
      className="c0"
    />
  </StyledComponent>
</Input>
`;

exports[`TestedComponent is Input and matches snapshot 3`] = `
.c0 {
  border: none;
  box-shadow: 0 4px #212529,0 -4px #212529,4px 0 #212529,-4px 0 #212529;
  font-family: 'Press Start 2P',monospace;
  line-height: calc(2 * 1.5em);
  margin: 4px;
  outline-color: "#e7e7e7";
  padding: 0.5rem 1rem;
  text-transform: capitalize;
  width: 100%;
}

@media (min-width:768px) {
  .c0 {
    font-size: 1.5em;
    line-height: 1.5em;
  }
}

<Input
  status={null}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Input-sc-1wcekai-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "border:none;box-shadow:0 4px ",
            [Function],
            ",0 -4px ",
            [Function],
            ",4px 0 ",
            [Function],
            ",-4px 0 ",
            [Function],
            ";font-family:",
            "'Press Start 2P', monospace",
            ";line-height:calc(2 * ",
            "1.5em",
            ");margin:4px;outline-color:",
            [Function],
            ";padding:0.5rem 1rem;text-transform:capitalize;width:100%;@media (min-width:",
            "768px",
            "){font-size:",
            "1.5em",
            ";line-height:",
            "1.5em",
            ";}",
          ],
        },
        "displayName": "Input",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Input-sc-1wcekai-0",
        "target": "input",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    status={null}
  >
    <input
      className="c0"
    />
  </StyledComponent>
</Input>
`;

exports[`TestedComponent is MessageBox and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "MessageBox-sc-1jj71pj-0",
        "isStatic": true,
        "rules": Array [
          "background-color:papayawhip;padding:1em;",
        ],
      },
      "displayName": "MessageBox",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "MessageBox-sc-1jj71pj-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is Modal and matches snapshot 1`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(0,0,0,0.4);
  display: none;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  overflow: auto;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
}

<Modal
  open={false}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Modal-sc-1lxhdwr-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "align-items:center;background-color:rgba(0,0,0,0.4);display:",
            [Function],
            ";height:100%;justify-content:center;left:0;overflow:auto;position:fixed;top:0;width:100%;z-index:1;",
          ],
        },
        "displayName": "Modal",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Modal-sc-1lxhdwr-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    open={false}
  >
    <div
      className="c0"
      open={false}
    />
  </StyledComponent>
</Modal>
`;

exports[`TestedComponent is Modal and matches snapshot 2`] = `
.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  background-color: rgba(0,0,0,0.4);
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  height: 100%;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  left: 0;
  overflow: auto;
  position: fixed;
  top: 0;
  width: 100%;
  z-index: 1;
}

<Modal
  open={true}
>
  <StyledComponent
    forwardedComponent={
      Object {
        "$$typeof": Symbol(react.forward_ref),
        "attrs": Array [],
        "componentStyle": ComponentStyle {
          "componentId": "Modal-sc-1lxhdwr-0",
          "isStatic": false,
          "lastClassName": "c0",
          "rules": Array [
            "align-items:center;background-color:rgba(0,0,0,0.4);display:",
            [Function],
            ";height:100%;justify-content:center;left:0;overflow:auto;position:fixed;top:0;width:100%;z-index:1;",
          ],
        },
        "displayName": "Modal",
        "foldedComponentIds": Array [],
        "render": [Function],
        "styledComponentId": "Modal-sc-1lxhdwr-0",
        "target": "div",
        "toString": [Function],
        "warnTooManyClasses": [Function],
        "withComponent": [Function],
      }
    }
    forwardedRef={null}
    open={true}
  >
    <div
      className="c0"
      open={true}
    />
  </StyledComponent>
</Modal>
`;

exports[`TestedComponent is PageWrapper and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "PageWrapper-sc-1xhm07a-0",
        "isStatic": true,
        "rules": Array [
          "margin:auto;width:96vw;@media (min-width:",
          "768px",
          "){width:80vw;}",
        ],
      },
      "displayName": "PageWrapper",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "PageWrapper-sc-1xhm07a-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;

exports[`TestedComponent is ThreeBlinking and matches snapshot 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "ThreeBlinking-sc-1k540pu-0",
        "isStatic": true,
        "rules": Array [
          "animation:",
          Keyframes {
            "id": "sc-keyframes-yGVCZ",
            "inject": [Function],
            "name": "yGVCZ",
            "rules": Array [
              "@-webkit-keyframes yGVCZ{0%{opacity:.2;}20%{opacity:1;}100%{opacity:.2;}}",
              "@keyframes yGVCZ{0%{opacity:.2;}20%{opacity:1;}100%{opacity:.2;}}",
            ],
            "toString": [Function],
          },
          " 1.4s linear infinite both;display:inline-block;::before{content:'.';}:nth-child(2){animation-delay:0.2s;}:nth-child(3){animation-delay:0.4s;}",
        ],
      },
      "displayName": "ThreeBlinking",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "ThreeBlinking-sc-1k540pu-0",
      "target": "div",
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
/>
`;
